### PR TITLE
add relations to models

### PR DESCRIPTION
**Pull request by patrickweh**
add columns to orders, contacts and vat_rates

add frontend fields for purchase-order

### CODE REVIEW

File: src/Http/Requests/CreateVatRateRequest.php [hunk 1]
```diff
@@ -14,6 +14,7 @@ public function rules(): array
         return [
             'uuid' => 'string|uuid|unique:vat_rates,uuid',
             'name' => 'required|string',
+            'footer_text' => 'string|nullable',
```
Comment (SirSplasch) [inline src/Http/Requests/CreateVatRateRequest.php:17]: move after rate_percentage

File: database/migrations/2023_09_24_115947_add_vat_rate_id_to_contacts_table.php
Comment (SirSplasch) [inline database/migrations/2023_09_24_115947_add_vat_rate_id_to_contacts_table.php:1]: migration name does not really match the content of the migration

File: resources/views/livewire/order/purchase.blade.php [hunk 2]
```diff
@@ -9,6 +9,37 @@
         </x-modal>
     </div>
 @endsection
+@section('state-card')
+    <x-card>
+        <div class="space-y-3">
+            <x-select
+                :label="__('Authorisation user')"
+                wire:model="order.approval_user_id"
```
Comment (SirSplasch) [inline resources/views/livewire/order/purchase.blade.php:17]: Doesn't this mean, that you can change the authorization user to yourself and than "approve/confirm" the order?

File: database/migrations/2023_09_24_115947_add_vat_rate_id_to_contacts_table.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                ->after('client_id')
+                ->nullable()
+                ->constrained('ledger_accounts')
+                ->onDelete('set null');
+            $table->foreignId('vat_rate_id')
+                ->after('expense_ledger_account_id')
+                ->nullable()
+                ->constrained('vat_rates')
+                ->onDelete('set null');
+            $table->string('vat_id')->after('credit_line')->nullable();
```
Comment (SirSplasch) [inline database/migrations/2023_09_24_115947_add_vat_rate_id_to_contacts_table.php:30]: Column named vat_id as string?

File: src/Http/Livewire/Order/Order.php [hunk 4]
```diff
@@ -62,6 +62,7 @@ public function mount(string $id = null): void
                 'addresses',
                 'client:id,name',
                 'contact.media',
+                'contact.bankConnections',
```
Comment (SirSplasch) [inline src/Http/Livewire/Order/Order.php:65]: Not sure if all columns of the bank_connections table are needed.

File: src/Actions/OrderPosition/CreateOrderPosition.php [hunk 5]
```diff
@@ -78,6 +79,10 @@ public static function models(): array
     public function performAction(): OrderPosition
     {
         $tags = Arr::pull($this->data, 'tags', []);
+        $order = Order::query()
+            ->with('orderType')
```
Comment (SirSplasch) [inline src/Actions/OrderPosition/CreateOrderPosition.php:83]: only id and order_type_enum are needed, or am I wrong?

**Review by SirSplasch — CHANGES_REQUESTED** [review]

File: database/migrations/2023_09_24_115947_add_vat_rate_id_to_contacts_table.php (hunk 3 above)
Comment (patrickweh) [reply database/migrations/2023_09_24_115947_add_vat_rate_id_to_contacts_table.php:30]: Umsatzsteuer ID like DE123456

**Review by SirSplasch — CHANGES_REQUESTED** [review]

**Review by SirSplasch — APPROVED** [review]

**Pull request merged**